robottelo/api/utils.py

```diff
@@ -825,6 +826,25 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self.temp.update(['locked'])
 
 
+@contextmanager
+def satelliteSetting(key_val: str):
+    """Context Manager to update the satellite setting and revert on exit
+
+    :param key_val: The setting name and value in format `setting_name=new_value`
+    """
+    name, value = key_val.split('=')
+    try:
+        setting = entities.Setting().search(query={'search': f'name={name.strip()}'})[0]
+    except IndexError:
+        raise KeyError(f'The setting {name} in not available in satellite.')
+    old_value = setting.value
+    setting.value = value.strip()
+    setting.update({'value'})
+    yield
+    setting.value = old_value
+    setting.update({'value'})
```

Comment:
what if it does not get update, will this affect other tests ? touching settings object would affect many other tests in case.

Reply:
Fixed with try catch here now! Thanks for the comment !